pydantic_ai_slim/pydantic_ai/models/fallback.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations as _annotations
+
+import sys
+from collections.abc import AsyncIterator
+from contextlib import AsyncExitStack, asynccontextmanager
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+from ..exceptions import ModelStatusError
+from . import KnownModelName, Model, ModelRequestParameters, StreamedResponse, infer_model
+
+if TYPE_CHECKING:
+    from ..messages import ModelMessage, ModelResponse
+    from ..settings import ModelSettings
+    from ..usage import Usage
+
+if sys.version_info < (3, 11):
+    from exceptiongroup import ExceptionGroup as ExceptionGroup
+else:
+    ExceptionGroup = ExceptionGroup
+
+
+@dataclass(init=False)
+class FallbackModel(Model):
+    """A model that uses one or more fallback models upon failure.
+
+    Apart from `__init__`, all methods are private or match those of the base class.
+    """
+
+    models: list[Model]
+
+    _model_name: str = field(repr=False)
+
+    def __init__(
+        self,
+        default_model: Model | KnownModelName,
+        *fallback_models: Model | KnownModelName,
+    ):
```

Comment:
Should it be parameter only?
```suggestion
        /
        ,
        default_model: Model | KnownModelName,
        *fallback_models: Model | KnownModelName,
    ):
```